vidyut-lipi/src/consts.tokens

```diff
@@ -0,0 +1,180 @@
+
```

Comment:
Let me see whether i understand this correctly - comment is represent data as rust data structures instead of JSON?
I don't quite understand the benefit - leaving aside mixing input data and code. 
If I still need to do a transform to hashmap - i am not saving any processing steps.
I don't understand the earlier comment - "Using native Rust data
structures instead will be faster (since there's no I/O) and simpler (since
there's no need for a heavy serde dependency)." 
- This is a simple flat json structure - no complex or deeply nested mappings. Should not tax any marshalling/unmarshalling library - unless of course serde is particularly bad. 
- with regards to IO - I/O cost is less than trivial irrespective of how it is delivered.

If the issue is having two extra files - regardless of the reason, they really aren't necessary. I just found it easy to use a script to generate them and then edit as necessary. Some one could just as easily copy the file, rename it and then edit it - same as could copy the mapping structure and edit. 

I am trying to understand what is gained by representing it as rust structure - i am surely missing something - i don't see any benefit.

Reply:
Sure, here are the benefits as they stand out to me.

Correctness:

1. We can make better use of Rust's powerful type system. We can model exactly which fields are used and which values those fields can take. We can likewise see where each value is used in the code, and linters can check whether any field values are unused or haven't been handled correctly. 

Speed:

2. Even simple I/O (~microseconds) and parsing (~microseconds) is orders of magnitude slower than accessing Rust datastructures in native code (~nanoseconds). One might object: "this I/O cost is trivial and incurred just once." Even so, I want to maintain correctness and speed in every aspect of the program.

Simplicity: 

3. The data here is fairly small. It is a closed set (the set of encoding schemes is small and bounded, and schemes rarely if ever change). It also doesn't have a clear application outside of the context of a transliteration library. Thus I think it's a great candidate to be defined directly in the code.

4. In general, I prefer not taking on new dependencies if we can avoid them. `serde` is a trustworthy library, but it is a relatively slow part of compilation, and its only use here is for converting JSON data to Rust structs.

Flexibility:

5. We can more easily build for WebAssembly, which means that we can use this transliterator in the frontend as well. WebAssembly generally doesn't have IO access, and it's easiest to include necessary data within the binary itself. More broadly, distribution in general becomes easier.